docs/learn/smart-contract-developers/deploy-up-with-lsp23.md

```diff
@@ -0,0 +1,447 @@
+
+```typescript
+async function main() {
+  // previous code
+
+  const lsp3DataKey = ERC725YDataKeys.LSP3.LSP3Profile;
+  const lsp3DataValue = ethers.randomBytes(32); // this is just a random value for the LSP3Profile
+}
+```
+
+#### Create the Universal Receiver data key
+
+```typescript
+async function main() {
+  // previous code
+
+  const universalReceiverDataKey =
+    ERC725YDataKeys.LSP1.LSP1UniversalReceiverDelegate;
+}
+```
+
+#### Create the permissions data keys and data values
+
+Let's start by creating the permissions data keys and data values of the Universal Receiver. The Universal Receiver will have the [`SUPER_SETDATA` and `REENTRANCY`](../../standards/universal-profile/lsp6-key-manager#permissions) permissions. At the moment we are writing this guide, the combination of these two permissions is `0x0000000000000000000000000000000000000000000000000000000000060080` however this may change in the future. You can check if this is still the case by looking at the [ERC725 Inspection Tool](https://erc725-inspect.lukso.tech/key-manager).
+
+```typescript
+async function main() {
+  // previous code
+
+  const universalReceiverPermissionsKey =
+    ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+    UniversalReceiverAddress.slice(2);
+  const universalReceiverPermissionsValue =
+    '0x0000000000000000000000000000000000000000000000000000000000060080'; // REENTRANCY & SUPER_SETDATA permissions
+}
+```
+
+Now we will want after deployment to have a main controller that has all the permissions on the Universal Profile. We will start by adding this address under our previous constants:
+
+```typescript
+const mainController = '0x3303Ce3b8644D566271DD2Eb54292d32F1458968'; // this is just a random address
+```
+
+Then we will create the data key and data value for the permissions of the main controller:
+
+```typescript
+async function main() {
+  // previous code
+
+  const mainControllerPermissionsKey =
+    ERC725YDataKeys.LSP6['AddressPermissions:Permissions'] +
+    mainController.slice(2);
+  const mainControllerPermissionsValue = ALL_PERMISSIONS; // main controller will have all permissions on the Universal Profile
+}
+```
+
+#### Create the AddressPermissions[] length data key and data value
+
+```typescript
+async function main() {
+  // previous code
+
+  const addressPermissionsLengthDataKey =
+    ERC725YDataKeys.LSP6['AddressPermissions[]'].length;
+
+  const abiCoder = new AbiCoder(); // instantiate the abiCoder
+  const addressPermissionsLengthDataValue = abiCoder.encode(['uint256'], [2]); // Address Permissions array length
+}
+```
+
+#### Create the AddressPermissions[] array data keys and data values
+
+```typescript
+async function main() {
+  // previous code
+
+  const addressPermissionsArrayFirstElementKey =
```

Comment:
Up to you but I would really recommend _erc725.js_ here for encoding this data key value pair. It will make it very easier.

This code below will generate the Array length + index

```js
const permissionData = erc725.encodeData([
  // the new list controllers addresses (= addresses with permissions set on the UP)
  // + the incremented `AddressPermissions[]` array length
  {
    keyName: 'AddressPermissions[]',
    value: [mainController, UniversalReceiverAddress],
  },
]);
```

https://docs.lukso.tech/learn/expert-guides/key-manager/grant-permissions

<img width="1056" alt="image" src="https://github.com/lukso-network/docs/assets/31145285/b1487652-7cf0-48b7-98b5-0fc3582a658c">

Reply:
thanks created a ticket to update this guide with erc725.lib